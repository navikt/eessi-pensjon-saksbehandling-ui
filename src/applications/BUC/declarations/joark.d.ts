
export interface JoarkFileVariant {
  variantformat:string;
  filnavn: string;
}

export interface JoarkDoc {
  dokumentInfoId: string;
  tittel: string;
  dokumentvarianter: Array<JoarkFileVariant>;
}
export interface JoarkPoster {
  journalpostId: string;
  tittel: string;
  tema: string;
  datoOpprettet: string;
  tilleggsopplysninger?: string;
  dokumenter: Array<JoarkDoc>
}

export interface JoarkFile {
  journalpostId: string;
  tittel: string;
  tema: string;
  datoOpprettet: Date;
  dokumentInfoId: string;
  tilleggsopplysninger?: string;
  variant: JoarkFileVariant
}

interface Content {
  base64: string;
}

export interface JoarkFileWithContent extends JoarkFile {
  content: Content;
  name: string;
  size: number;
  mimetype: string;
}
