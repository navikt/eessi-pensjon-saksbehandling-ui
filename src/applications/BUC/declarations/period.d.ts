export interface PeriodDate {
  day: string;
  month: string;
  year: string;
}

export interface Country {
  label: string;
  value: string;
}

export interface Period {
  id?: number,
  type: string;
  startDate: PeriodDate;
  endDate?: PeriodDate;
  dateType: string;
  uncertainDate: boolean;
  country: Country;
  comment: string;
  otherType?: string;
  insuranceName?: string;
  insuranceType?: string;
  insuranceId ?: string;
  workActivity ?: string;
  workName ?: string;
  workType ?: string;
  workStreet ?: string;
  workCity ?: string;
  workZipCode ?: string;
  workRegion ?: string;
  childFirstName ?: string;
  childLastName ?: string;
  childBirthDate ?: PeriodDate;
  learnInstitution ?: string;
  payingInstitution ?: string;
  attachments ?: Array[any];
  [key: string]: any;
}

export type Periods = Array<Period>

export type PeriodErrors = {[k: string]: string | undefined}

export interface PayloadPeriod {
  land: string;
  periode: null | {
    lukketPeriode?: {
      fom: string | null;
      tom: string | null;
    }
    openPeriode?:  {
      fom: string | null;
      extra: string;
    }
  };
  vedlegg ?: Array[any];
  trygdeordningnavn ?: string;
  medlemskap ?: string;
  forsikkringEllerRegistreringNr ?: string;
  annenInformasjon ?: string;
  usikkerDatoIndikator ?: string;
  jobbUnderAnsattEllerSelvstendig ?: string;
  navnFirma ?: string;
  typePeriode ?: string;
  adresseFirma ?: {
    postnummer: string | undefined;
    by: string | undefined;
    land: string;
    gate: string | undefined;
    region: string | undefined;
  };
  navnPaaInstitusjon ?: string;
  informasjonBarn ?:  {
    etternavn: string | null | undefined;
    fornavn: string | null | undefined;
    foedseldato: string | null | undefined;
    land: string;
  }
}

export interface Payload {
  ansattSelvstendigPerioder ?: Array<PayloadPeriod>;
  boPerioder ?: Array<PayloadPeriod>;
  barnepassPerioder ?: Array<PayloadPeriod>;
  frivilligPerioder ?: Array<PayloadPeriod>;
  forsvartjenestePerioder ?: Array<PayloadPeriod>;
  foedselspermisjonPerioder ?: Array<PayloadPeriod>;
  opplaeringPerioder ?: Array<PayloadPeriod>;
  arbeidsledigPerioder ?: Array<PayloadPeriod>;
  sykePerioder ?: Array<PayloadPeriod>;
  andrePerioder ?: Array<PayloadPeriod>;
}

export interface P4000Info {
  person: any,
  bank: any,
  stayAbroad: Array<Period>
}

export interface P4000PayloadInfo {
  personInfo: any,
  bankInfo: any,
  periodeInfo: Array<PayloadPeriod>
}
