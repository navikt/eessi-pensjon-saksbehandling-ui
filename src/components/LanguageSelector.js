import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators }  from 'redux';
import { Select } from 'nav-frontend-skjema';
import { translate } from 'react-i18next';

import * as languageActions from '../actions/language';

const mapStateToProps = (state) => {
  return {
    language : state.ui.language
  };
};

const mapDispatchToProps = (dispatch) => {
  return {actions: bindActionCreators(Object.assign({}, languageActions), dispatch)};
};

class LanguageSelector extends Component {

  changeLanguage(e) {

    let { actions } = this.props;
    actions.changeLanguage(e.target.value);
  }

  render () {

    let { t, language } = this.props;
    return <Select label={t('chooseLanguage')} value={language} onChange={this.changeLanguage.bind(this)}>
      <option value='en'>English</option>
      <option value='nb'>Bokm√•l</option>
      <option value='nn'>Nynorsk</option>
    </Select>
  }
}

LanguageSelector.propTypes = {
  language : PropTypes.obj,
  t        : PropTypes.func,
  actions  : PropTypes.obj
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(
  translate()(LanguageSelector)
);

