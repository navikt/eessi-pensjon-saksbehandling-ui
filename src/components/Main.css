@font-face {
   font-family: "Source Sans Pro";
   src: url(../fonts/SourceSansPro-Regular.ttf)
}

@font-face {
   font-family: "Source Sans Pro Bold";
   src: url(../fonts/SourceSansPro-Bold.ttf)
}

h1 {
   font-weight: bold !important;
   font-size: 2.5 rem !important;
}

h5 {
   font-size: 1.25 rem !important;
}

button,
input,
optgroup,
select,
textarea {
    font-family: sans-serif;
    font-size: 100%;
    line-height: 1.15;
    margin: 0px;
}

button,
input {
    overflow: visible;
}

button,
select {
    text-transform: none;
}

button,
html [type="button"],
[type="reset"],
[type="submit"] {
    -webkit-appearance: button;
}

* {
    font-family: "Source Sans Pro";
    box-sizing: border-box;
}

input,
button,
select,
textarea {
    font-family: inherit;
    font-size: inherit;
    line-height: inherit;
}

.knapp {
    display: inline-block;
    font-weight: 600;
    font-family: "Source Sans Pro", Arial, sans-serif;
    font-size: 1em;
    white-space: nowrap;
    border-radius: 0.125em;
    cursor: pointer;
    user-select: none;
    letter-spacing: 0.0625em;
    line-height: 1;
    text-transform: uppercase;
    text-decoration: none;
    background-color: transparent;
    padding: calc(0.5em - 2px) calc(2.5em - 2px) calc(0.625em - 2px);
    border: 2px solid rgb(0, 103, 197);
    color: rgb(0, 103, 197);
}

.skjemaelement {
    margin-bottom: 1rem;
}

.skjemaelement__label {
    font-family: "Source Sans Pro", Arial, sans-serif;
    font-size: 1rem;
    font-weight: 400;
    display: block;
    line-height: 1.4;
    margin-bottom: 0.5rem;
}

.input--fullbredde {
    width: 100%;
}

.skjemaelement__input {
    padding: 0.45rem;
    background-color: rgb(255, 255, 255);
    border-radius: 4px;
    border: 1px solid rgb(183, 177, 169);
    box-sizing: border-box;
}

.alertstripe {
    background-color: rgb(255, 255, 255);
    padding: 1rem;
    display: flex;
    align-content: center;
    border-radius: 2px;
    color: rgb(62, 56, 50);
}

.alertstripe--suksess {
    border: 1px solid rgb(6, 137, 58);
}

.alertstripe--advarsel {
    border: 1px solid rgb(186, 58, 38);
}

.alertstripe__ikon {
    margin-right: 0.75rem;
    padding-right: 0.75rem;
    width: 2.5rem;
    display: flex;
    flex-shrink: 0;
    align-items: center;
}

.alertstripe:not(.alertstripe--solid) .alertstripe__ikon {
    border-right: 1px solid rgb(183, 177, 169);
}

svg:not(:root) {
    overflow: hidden;
}

.typo-normal {
    margin: 0px;
    font-family: "Source Sans Pro", Arial, sans-serif;
    font-size: 1rem;
    line-height: 1.375rem;
    font-weight: 400;
}

.typo-ingress,
.typo-normal {
    max-width: 640px;
}

.alertstripe .alertstripe__tekst {
    margin: auto 0px;
    max-width: 37.5rem;
}

.spinner {
    display: inline-block;
    position: relative;
    animation: rotering 2s linear infinite;
}

.spinner--m {
    height: 32px;
    width: 32px;
    background-size: 32px 32px;
    animation-duration: 1.1s;
}
