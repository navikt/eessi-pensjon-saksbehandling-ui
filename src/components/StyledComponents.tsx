import styled from 'styled-components'
import {Panel, RadioGroup} from '@navikt/ds-react'
import { PaddedHorizontallyDiv } from '@navikt/hoykontrast'

export const OneLineSpan = styled.span`
  white-space: nowrap;
`
export const SeparatorSpan = styled.span`
  margin-left: 0.25rem;
  margin-right: 0.25rem;
`
export const SpinnerDiv = styled.div`
  position: absolute;
  top: 3rem;
  left: 0;
  right: 0;
`
export const HorizontalLineSeparator = styled.div`
  height: 1px;
  background: linear-gradient(90deg,
    var(--a-bg-subtle) 0%,
    var(--a-border-strong) 5%,
    var(--a-border-strong) 95%,
    var(--a-bg-subtle) 100%
  );
  width: -webkit-fill-available;
  margin-left: 2rem;
  margin-right: 2rem;
`
export const WithErrorPanel = styled(Panel)`
  padding: 0rem;
  background-color: transparent;
  border: none;
  &.error {
    margin: -4px;
    border: 4px solid var(--a-border-danger) !important;
  }
`

export const RepeatableRow = styled(PaddedHorizontallyDiv)`
  padding-top: 0.3rem;
  padding-bottom: 0.3rem;
  min-height: 3.0rem;
  &.new {
    background-color: rgba(236, 243, 153, 0.5);
  };
  &.error {
    background-color: rgba(255, 0, 0, 0.2);
  };
  &:hover:not(.new):not(.error) {
    background-color: var(--a-gray-100);
  }
  &:not(:hover) .control-buttons {
    position: absolute;
    margin-left: -10000px;
  }
`

export const RepeatableRowAlternateColors = styled(RepeatableRow)`
  border: 1px solid transparent;
  &:nth-child(even){
    background-color: var(--a-gray-300);
  }
   &:hover:not(.new):not(.error) {
    background-color: var(--a-gray-100);
    border: 1px solid #000000;
  }
`


export const RepeatableRowNoHorizontalPadding = styled(RepeatableRow)`
  padding-left:0;
  padding-right:0;
`

export const Hr = styled.div`
   background: var(--a-border-default);
   width: 100%;
   height: 1px;
`

export const SpacedHr = styled(Hr)`
   margin-top: 0.5rem;
   margin-bottom: 0.5rem;
`

export const HorizontalRadioGroup = styled(RadioGroup)`
  > .navds-radio-buttons {
    display: flex;
    gap: var(--a-spacing-4);
    margin-bottom: var(--a-spacing-2)
  }
`

