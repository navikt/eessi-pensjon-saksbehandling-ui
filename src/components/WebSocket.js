import React, { useState, useEffect } from 'react'
import {connectToWebSocket} from '../utils/websocket'
import { connect } from '../store'

const logData = (e) => console.log(e.data)

const mapStateToProps = (state) => {
  return {
    aktoerId: state.app.params.aktoerId,
    avdodfnr: state.app.params.avdodfnr
  }
}


const WebSocket = () => {
  const [connection, setConnection] = useState(null)
  const [ready, setReady] = useState(false)


  useEffect( () => {
      let websocketConnection = connectToWebSocket(()=>setReady(true), logData, logData, logData)
      setConnection(websocketConnection)
    }, []
  )

  useEffect( () => {
    if(ready){
      let message = {subscriptions: [aktoerId, avdodfnr].filter(el=>el)}
      connection.send(JSON.stringify(message))
    }

  }, [ready, aktoerId, avdodfnr])

  return <div className='WebSocket'></div>

}

export default connect(mapStateToProps, ()=>{})(WebSocket)
