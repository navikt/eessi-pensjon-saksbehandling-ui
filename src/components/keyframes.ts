import { keyframes } from 'styled-components'

export const slideInFromRight = keyframes`
  0% {
    opacity: 0;
    transform: translateX(20px);
  }
  100% {
    opacity: 1;
    transform: translateX(0);
  }
`
export const slideInFromLeft = keyframes`
  0% {
    opacity: 0;
    transform: translateX(-20px);
  }
  100% {
    opacity: 1;
    transform: translateX(0);
  }
`
export const animationOpen = keyframes`
  0% {
    height: 0%;
    max-height: 0;
  }
  100% {
    max-height: 150em;
    height: 100%;
  }
`
export const animationClose = keyframes`
  0% {
    max-height: 150em;
    height: 100%;
  }
  100% {
    max-height: 0;
    height: 0%;
  }
`
export const rotating = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`
export const fadeIn = keyframes`
  0% { opacity: 0; }
  50% { opacity: 1; }
  100% { opacity: 1; }
`
export const fadeOut = keyframes`
  0% { opacity: 1; }
  50% { opacity: 1; }
  100% { opacity: 0; }
`
export const progressBarAnimation = keyframes`
  0% {background-position: 1rem 0;} to {background-position: 0 0;}
`
