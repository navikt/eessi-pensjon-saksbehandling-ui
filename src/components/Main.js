import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './Main.css';
import 'bootstrap/dist/css/bootstrap.min.css';

import { translate } from 'react-i18next';

import * as Nav from './Nav'
import TopHeader from './TopHeader';

class Main extends Component {

  render () {

    let { t } = this.props;

    return <div className='h-100'>
      <TopHeader/>
      <Nav.Container>
        <Nav.Row>
          <Nav.Column className='py-3 text-left'>
            <h1 className='mt-3 appTitle'>{t('content:pageTitle')}</h1>
          </Nav.Column>
        </Nav.Row>
        <Nav.Row>
          <Nav.Column className='py-3 text-left'>
            {this.props.children}
          </Nav.Column>
        </Nav.Row>
      </Nav.Container>
    </div>
  }
}

Main.propTypes = {
  children : PropTypes.node.isRequired,
  t        : PropTypes.func.isRequired
};

export default translate()(Main);
