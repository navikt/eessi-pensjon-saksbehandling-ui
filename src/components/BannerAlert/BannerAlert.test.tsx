import { AlertVariant } from 'src/declarations/components'
import { screen, render } from '@testing-library/react'
import BannerAlert, { BannerAlertProps } from './BannerAlert'

describe('src/components/Alert/Alert', () => {
  const initialMockProps: BannerAlertProps = {
    variant: 'error',
    message: 'mockErrorMessage',
    error: undefined,
    onClose: jest.fn()
  }

  it('Render: match snapshot', () => {
    const { container } = render(<BannerAlert {...initialMockProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Render: has proper HTML structure', () => {
    render(<BannerAlert {...initialMockProps} />)
    expect(screen.getByRole('alert')).toBeInTheDocument()
    expect(screen.getByRole('alert')).toHaveTextContent('mockErrorMessage')
  })

  it('Render: has proper HTML structure with error message', () => {
    render(<BannerAlert {...initialMockProps} error='mockError' />)
    expect(screen.getByRole('alert')).toHaveTextContent('mockErrorMessage: mockError')
  })

  it('Render: has proper HTML structure as client in OK type', () => {
    render(<BannerAlert {...initialMockProps} variant='success'/>)
    expect(screen.getByRole('alert').getAttribute('class')).toMatch(/status-success/gi)
  })

  it('Render: has proper HTML structure as client in WARNING type', () => {
    render(<BannerAlert {...initialMockProps} variant='warning' />)
    expect(screen.getByRole('alert').getAttribute('class')).toMatch(/status-warning/gi)
  })

  it('Render: has proper HTML structure as client in ERROR type', () => {
    render(<BannerAlert {...initialMockProps} variant='error' />)
    expect(screen.getByRole('alert').getAttribute('class')).toMatch(/status-error/gi)
  })

  it('Render: Pretty prints a error message', () => {
    const error = {
      status: 'error' as AlertVariant,
      message: 'message',
      error: 'error',
      uuid: 'uuid'
    }
    render(<BannerAlert {...initialMockProps} error={error} />)
    expect(screen.getByRole('alert')).toHaveTextContent('mockErrorMessage: message - error - uuid')
  })

  it('Render: Pretty prints a string error', () => {
    const error = 'error'
    render(<BannerAlert {...initialMockProps} error={error} />)
    expect(screen.getByRole('alert')).toHaveTextContent('mockErrorMessage: error')
  })

  it('Render: Close button to be in the document', () => {
    render(<BannerAlert {...initialMockProps} variant='error' />);
    expect(screen.getByTestId('c-alert--close-icon')).toBeInTheDocument();
  })
})
