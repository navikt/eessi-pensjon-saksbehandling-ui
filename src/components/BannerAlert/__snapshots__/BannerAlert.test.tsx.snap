// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Alert/Alert Render: match snapshot 1`] = `
.c1 {
  opacity: 0;
  -webkit-animation: kGLwuF 1s forwards;
  animation: kGLwuF 1s forwards;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 10;
  width: 100%;
  border-radius: 0px !important;
  border: 0px !important;
}

.c1 .alertstripe--tekst {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: none !important;
}

ui-fss.g40[id="c0"] {
  content: "c1,";
}

.c3 {
  position: absolute;
  top: 0.25rem;
  right: 0.25rem;
  cursor: pointer;
}

ui-fss.g41[id="c2"] {
  content: "c3,";
}

<div
  class="c0 c1 status-error navds-alert navds-alert--error navds-alert--medium"
  role="alert"
>
  <svg
    aria-labelledby="title-:r0:"
    class="navds-alert__icon"
    fill="none"
    focusable="false"
    height="1em"
    role="img"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="title-:r0:"
    >
      Feil
    </title>
    <path
      d="M11.999 0C5.395 0 .013 5.372 0 11.976a11.923 11.923 0 0 0 3.498 8.493A11.925 11.925 0 0 0 11.977 24H12c6.603 0 11.986-5.373 12-11.978C24.012 5.406 18.64.012 11.999 0Z"
      fill="#BA3A26"
    />
    <path
      d="m12 10.651 3.372-3.372a.954.954 0 1 1 1.349 1.35L13.349 12l3.372 3.372a.954.954 0 1 1-1.35 1.349L12 13.349 8.628 16.72a.954.954 0 1 1-1.349-1.35L10.651 12 7.28 8.628A.954.954 0 1 1 8.63 7.28L12 10.651Z"
      fill="#fff"
    />
  </svg>
  <div
    class="navds-alert__wrapper navds-body-long"
  >
    mockErrorMessage
    <svg
      class="c2 c3"
      data-testid="c-alert--close-icon"
      fill="none"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0Zm3.571 7L17 8.429 13.428 12 17 15.571 15.571 17 12 13.428 8.429 17 7 15.571 10.572 12 7 8.429 8.429 7 12 10.572 15.571 7Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;
